flockoff/validator/database.py

```
@@ -1,57 +1,123 @@
+                         (uid INTEGER, hotkey TEXT, score REAL, PRIMARY KEY (uid, hotkey))"""
+            )
+
+            c.execute(
+                """CREATE TABLE IF NOT EXISTS dataset_revisions
+                           (namespace TEXT PRIMARY KEY, revision TEXT)"""
+            )
+
+            self.conn.commit()
+        except sqlite3.Error as e:
+            logger.error(f"Failed to initialize database tables: {str(e)}")
+            raise DatabaseError(f"Failed to create database tables: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)